pgcli/key_bindings.py

```diff
@@ -102,9 +102,11 @@ def _(event):
     # history search, and one of several conditions are True
     @kb.add(
         "enter",
-        filter=~(has_completions | is_searching) & buffer_should_be_handled(pgcli),
+        filter=~(completion_is_selected | is_searching)
```

Comment:
I recall using `has_completions` rather than `completion_is_selected` in #1109 consciously, but I don't remember _why_. Can you give an example key-press sequence to show where the existing behaviour causes issues? I use multi-line and vim mode, so it's quite possible that I didn't fully test the emacs/single-line mode - apologies for that.

Reply:
I also tried to add a feature test for keypress handling in #1109, but ran into issues (I was having trouble sending the correct keypress sequences as I recall). 

It'd be nice if we could add some tests in this area, as I think it's important to preserve a consistent UI. Do you know if anyone has tried before - or whether it's even possible to test the key press handling such as this in a feature test?

Reply:
Reproducible key sequence:

Ensure multiline mode is OFF either via the config file or by toggling it via F3 key. 

SELECT a<enter>

When you type `SELECT a` it will pop up a completion menu with possible suggestions for the letter `a`. If you press enter it should submit the command to the server but instead a newline is inserted with the completion menu kept open. 

This PR fixes that situation.